[UnitTest] AttributeError: module 'numpy' has no attribute 'ComplexWarning'

## 🐛 Bug

<!-- A clear and concise description of what the bug is. -->
```
================================================ short test summary info ================================================
FAILED tests/python/common/sampling/test_sampling.py::test_global_uniform_negative_sampling[int32] - AttributeError: module 'numpy' has no attribute 'ComplexWarning'
FAILED tests/python/common/sampling/test_sampling.py::test_global_uniform_negative_sampling[int64] - AttributeError: module 'numpy' has no attribute 'ComplexWarning'
========================== 2 failed, 2872 passed, 60 skipped, 42 warnings in 148.98s (0:02:28) ==========================
```
## To Reproduce

Steps to reproduce the behavior:

1. `pip install  dgl -f https://data.dgl.ai/wheels/torch-2.4/cu121/repo.html`
2. `PYTHONPATH=tests:$PYTHONPATH python -m pytest tests/python/common/`

```
dgl                           2.4.0+cu121
numpy                         2.1.0
torch                         2.4.0+cu121
torch_geometric               2.5.3
torcheval                     0.0.7
torchmetrics                  1.4.1
```
<!-- If you have a code sample, error messages, stack traces, please provide it here as well -->

## Expected behavior

<!-- A clear and concise description of what you expected to happen. -->

## Environment

 - DGL Version (e.g., 1.0):
 - Backend Library & Version (e.g., PyTorch 0.4.1, MXNet/Gluon 1.3):
 - OS (e.g., Linux):
 - How you installed DGL (`conda`, `pip`, source):
 - Build command you used (if compiling from source):
 - Python version:
 - CUDA/cuDNN version (if applicable):
 - GPU models and configuration (e.g. V100):
 - Any other relevant information:

## Additional context

<!-- Add any other context about the problem here. -->

I got three more FAILED things when trying to reproduce the behavior on a freshly opened device. I'll take a look at them as well.

```
FAILED tests/python/common/data/test_data.py::test_as_nodepred1 - ModuleNotFoundError: No module named 'rdflib'
FAILED tests/python/common/data/test_data.py::test_as_nodepred2 - ModuleNotFoundError: No module named 'rdflib'
FAILED tests/python/common/data/test_utils.py::test_add_nodepred_split - ModuleNotFoundError: No module named 'rdflib'
```

--- a/tests/python/common/sampling/test_sampling.py
+++ b/tests/python/common/sampling/test_sampling.py
@@ -1774,7 +1774,7 @@ def test_sample_neighbors_exclude_edges_homoG(dtype, fused):
 
 @pytest.mark.parametrize("dtype", ["int32", "int64"])
 def test_global_uniform_negative_sampling(dtype):
-    # warnings.simplefilter("ignore", np.ComplexWarning)
+    warnings.simplefilter("ignore", np.exceptions.ComplexWarning)
     g = dgl.graph(([], []), num_nodes=1000).to(F.ctx())
     src, dst = dgl.sampling.global_uniform_negative_sampling(
         g, 2000, False, True

--- a/tests/python/common/sampling/test_sampling.py
+++ b/tests/python/common/sampling/test_sampling.py
@@ -1774,7 +1774,7 @@ def test_sample_neighbors_exclude_edges_homoG(dtype, fused):
 
 @pytest.mark.parametrize("dtype", ["int32", "int64"])
 def test_global_uniform_negative_sampling(dtype):
-    warnings.simplefilter("ignore", np.ComplexWarning)
+    # warnings.simplefilter("ignore", np.ComplexWarning)
     g = dgl.graph(([], []), num_nodes=1000).to(F.ctx())
     src, dst = dgl.sampling.global_uniform_negative_sampling(
         g, 2000, False, True

